packages/block-editor/src/components/link-control/index.js

```diff
@@ -119,9 +119,11 @@ function LinkControl( {
 		);
 	};
 
-	const handleEntitySearch = async ( val ) => {
+	const handleEntitySearch = async ( val, args ) => {
 		const results = await Promise.all( [
-			fetchSearchSuggestions( val ),
+			fetchSearchSuggestions( val, {
+				...( args.isInitialSuggestions ? { perPage: 3 } : {} ),
```

Comment:
Can we not assume it's an initial suggestion when `val` is an empty string?

Reply:
Yes we can. Now we have the refactor we can remove a lot of the logic and just rely on the `input` value. Thanks @obenland!

Reply:
Seems like `isInitialSuggestions` is still around. Or did you mean it can be refactored once this is in?